sql/core/src/main/scala/org/apache/spark/sql/internal/CatalogImpl.scala

```diff
@@ -262,11 +262,13 @@ class CatalogImpl(sparkSession: SparkSession) extends Catalog {
       throw new AnalysisException("Cannot create hive serde table with createExternalTable API.")
     }
 
+    val location = new CaseInsensitiveMap(options).get("path")
+    val storageProps = options.filterNot { case (k, _) => k.toLowerCase == "path" }
```

Comment:
I saw this two line duplicated in at least three places. Maybe we should create a util to extract the path from options?